openedx/features/subscriptions/models.py

```diff
@@ -106,3 +108,67 @@ def get_valid_subscriptions(self, site, username=None):
             logger.warning('Valid subscription count exceeds 1.')
 
         return user_subscriptions.filter(subscription_id__in=valid_user_subscriptions_ids).order_by('-created')
+
+
+class UserSubscriptionHistory(TimeStampedModel):
+    """
+    """
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    subscription_id = models.PositiveIntegerField(db_index=True)
+    expiration_date = models.DateField(default=None, null=True, blank=True)
+
+    LIMITED_ACCESS = 'limited-access'
+    FULL_ACCESS_COURSES = 'full-access-courses'
+    FULL_ACCESS_TIME_PERIOD = 'full-access-time-period'
+    LIFETIME_ACCESS = 'lifetime-access'
+
+    SUBSCRIPTION_TYPE_CHOICES = (
+        (LIMITED_ACCESS, 'Limited Access', ),
+        (FULL_ACCESS_COURSES, 'Full Access (Courses)', ),
+        (FULL_ACCESS_TIME_PERIOD, 'Full Access (Time period)', ),
+        (LIFETIME_ACCESS, 'Lifetime Access', ),
+    )
+
+    subscription_type = models.CharField(choices=SUBSCRIPTION_TYPE_CHOICES, max_length=30)
+    max_allowed_courses = models.PositiveIntegerField(
+        help_text=_(
+            'Maximum number of courses allowed to enroll in paid tracks using the subscription.'
+        ),
+        null=True,
+        blank=True
+    )
+    course_enrollments = models.ManyToManyField(CourseEnrollment, blank=True)
+    site = models.ForeignKey(Site, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return "Subscription of type {subscription_type} with ID {subscription_id} for User {user_email}".format(
+            subscription_type=self.subscription_type,
+            subscription_id=self.subscription_id,
+            user_email=self.user.username,
+        )
+
+
+    class Meta:
+        get_latest_by = 'modified'
+        ordering = ('-modified', '-created',)
+
+@receiver(post_save, sender=UserSubscription)
+def update_user_subscription_history(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    Add user subscription changes to user subscription history.
+
+    Args:
```

Comment:
Update the docstrings according to edX standards: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

Reply:
done